src/PubSub/V1/Gapic/PublisherGapicClient.php

```diff
@@ -247,49 +247,26 @@ public static function parseName($formattedName, $template = null)
-     *           optional argument is specified, then this option is unused.
-     *     @type CredentialsLoader $credentialsLoader
-     *           A CredentialsLoader object created using the Google\Auth library.
+     *     Options for configuring the service API wrapper.
+     *
+     *     @type string $serviceAddress The address of the API remote host.
+     *           Default 'pubsub.googleapis.com:443'.
      *     @type string[] $scopes A string array of scopes to use when acquiring credentials.
-     *                          Defaults to the scopes for the Google Cloud Pub/Sub API.
+     *           Defaults to the scopes for the Google Cloud Pub/Sub API.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
      *     @type string $clientConfigPath
      *           Path to a JSON file containing client method configuration, including retry settings.
      *           Specify this setting to specify the retry behavior of all methods on the client.
      *           By default this settings points to the default client config file, which is provided
-     *           in the resources folder. The retry settings provided in this option can be overridden
-     *           by settings in $retryingOverride
-     *     @type array $retryingOverride
-     *           An associative array in which the keys are method names (e.g. 'createFoo'), and
-     *           the values are retry settings to use for that method. The retry settings for each
-     *           method can be a {@see Google\ApiCore\RetrySettings} object, or an associative array
-     *           of retry settings parameters. See the documentation on {@see Google\ApiCore\RetrySettings}
-     *           for example usage. Passing a value of null is equivalent to a value of
-     *           ['retriesEnabled' => false]. Retry settings provided in this setting override the
-     *           settings in $clientConfigPath.
-     *     @type callable $authHttpHandler A handler used to deliver PSR-7 requests specifically
-     *           for authentication. Should match a signature of
-     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
-     *     @type callable $httpHandler A handler used to deliver PSR-7 requests. Should match a
-     *           signature of `function (RequestInterface $request, array $options) : PromiseInterface`.
-     *           NOTE: This option is only valid when utilizing the REST transport.
+     *           in the resources folder.
+     *     @type bool $disableRetries Determines whether or not retries defined
```

Comment:
The rationale for adding this and removing `retryingOverride` was to keep things a little simpler and just provide one recommended way for users to override client configurations. Additionally, if users want to specify just one method to override they can already do so at the method call level with the `retrySettings` option.

Reply:
This seems fine. However, what about in addition to `clientConfigPath` we also allow a `clientConfig` setting, which is a PHP array instead of a json file (like the distinction between `keyFile` and `keyFilePath`. Otherwise the only way to adjust retry settings on all calls is to define a json file and point to it - it would be good to allow this from code.

Reply:
I like that idea, sounds good to me.